### ISSUE TITLE
jtable scrolling mouse

### ISSUE THREAD

**Issue by oliviermorpa** [mgarin/weblaf#211]
good pm,
when scrolling mouse to go down/up in stable, it's very slowly.
When we click to go down/up. it's ok. only scrolling.

when using regular jtable, its ok.
Have a nice day

**Comment by oliviermorpa** [comment]
any news ?

here is an exemple:

import java.awt.BorderLayout;

import javax.swing.JScrollPane;
import javax.swing.SwingUtilities;

import com.alee.laf.WebLookAndFeel;
import com.alee.laf.rootpane.WebFrame;
import com.alee.laf.table.WebTable;


public class Main {

	public static void main(final String[] args) throws Exception {
	
		SwingUtilities.invokeLater(new Runnable() {
			public void run() {
				            WebLookAndFeel.initializeManagers ();

		        
					
			ExampleFrame exampleFrame = new ExampleFrame();
			
			
			
			exampleFrame.setVisible(true);
			}});

	}
}
 class ExampleFrame extends WebFrame {

    /**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
public ExampleFrame() {
	super();
	setDefaultCloseOperation(WebFrame.DISPOSE_ON_CLOSE);
	
	
	String[] columnNames = {"First Name",
            "Last Name",
            "Sport",
            "# of Years",
            "Vegetarian"};
	
        
	Object[][] data = {
			{"Kathy", "Smith",
			     "Snowboarding", new Integer(5), new Boolean(false)},
			    {"John", "Doe",
			     "Rowing", new Integer(3), new Boolean(true)},
			    {"Sue", "Black",
			     "Knitting", new Integer(2), new Boolean(false)},
			    {"Jane", "White",
			     "Speed reading", new Integer(20), new Boolean(true)},
			    {"Joe", "Brown",
			     "Pool", new Integer(10), new Boolean(false), "Kathy", "Smith",
				     "Snowboarding", new Integer(5), new Boolean(false)},
					    {"John", "Doe",
					     "Rowing", new Integer(3), new Boolean(true)},
					    {"Sue", "Black",
					     "Knitting", new Integer(2), new Boolean(false)},
					    {"Jane", "White",
					     "Speed reading", new Integer(20), new Boolean(true)},
					    {"Joe", "Brown",
					     "Pool", new Integer(10), new Boolean(false), "Kathy", "Smith",
						     "Snowboarding", new Integer(5), new Boolean(false)},
							    {"John", "Doe",
							     "Rowing", new Integer(3), new Boolean(true)},
							    {"Sue", "Black",
							     "Knitting", new Integer(2), new Boolean(false)},
							    {"Jane", "White",
							     "Speed reading", new Integer(20), new Boolean(true)},
							    {"Joe", "Brown",
							     "Pool", new Integer(10), new Boolean(false), "Kathy", "Smith",
								     "Snowboarding", new Integer(5), new Boolean(false)},
									    {"John", "Doe",
									     "Rowing", new Integer(3), new Boolean(true)},
									    {"Sue", "Black",
									     "Knitting", new Integer(2), new Boolean(false)},
									    {"Jane", "White",
									     "Speed reading", new Integer(20), new Boolean(true)},
									    {"Joe", "Brown",
									     "Pool", new Integer(10), new Boolean(false)}, {"Kathy", "Smith",
		     "Snowboarding", new Integer(5), new Boolean(false)},
		    {"John", "Doe",
		     "Rowing", new Integer(3), new Boolean(true)},
		    {"Sue", "Black",
		     "Knitting", new Integer(2), new Boolean(false)},
		    {"Jane", "White",
		     "Speed reading", new Integer(20), new Boolean(true)},
		    {"Joe", "Brown",
		     "Pool", new Integer(10), new Boolean(false), "Kathy", "Smith",
			     "Snowboarding", new Integer(5), new Boolean(false)},
				    {"John", "Doe",
				     "Rowing", new Integer(3), new Boolean(true)},
				    {"Sue", "Black",
				     "Knitting", new Integer(2), new Boolean(false)},
				    {"Jane", "White",
				     "Speed reading", new Integer(20), new Boolean(true)},
				    {"Joe", "Brown",
				     "Pool", new Integer(10), new Boolean(false), "Kathy", "Smith",
					     "Snowboarding", new Integer(5), new Boolean(false)},
						    {"John", "Doe",
						     "Rowing", new Integer(3), new Boolean(true)},
						    {"Sue", "Black",
						     "Knitting", new Integer(2), new Boolean(false)},
						    {"Jane", "White",
						     "Speed reading", new Integer(20), new Boolean(true)},
						    {"Joe", "Brown",
						     "Pool", new Integer(10), new Boolean(false), "Kathy", "Smith",
							     "Snowboarding", new Integer(5), new Boolean(false)},
								    {"John", "Doe",
								     "Rowing", new Integer(3), new Boolean(true)},
								    {"Sue", "Black",
								     "Knitting", new Integer(2), new Boolean(false)},
								    {"Jane", "White",
								     "Speed reading", new Integer(20), new Boolean(true)},
								    {"Joe", "Brown",
								     "Pool", new Integer(10), new Boolean(false)}
		};	
	WebTable table = new WebTable(data, columnNames);
	JScrollPane scrollPane = new JScrollPane(table);
	table.setFillsViewportHeight(true);
	
    setLayout(new BorderLayout());

    getContentPane().add(scrollPane, BorderLayout.CENTER);
   
    setSize(600,200);
}


}

**Comment by mgarin** [comment]
Unfortunately I am not able to reproduce this issue anyhow. I tried your example with `WebLookAndFeel.initializeManagers ();` replaced by `WebLookAndFeel.install ();` to check table with WebLaF UI - it never "lags". I also tried my own examples but got the same results with large tables.

I tried almost all common options:
- Win 7/8
- JDK 6(u30/u45)/7/8(u20/25)
- Both 32 and 64 -bit JDK versions
- WebLaF v1.28 and v1.29 pre-release

Also tried:
- Wheel scroll
- Keys scroll
- Dragging scrollbar

And all the same - table scroll works just fine in all cases on all JDK/library versions.

Could you try running this example with pre-release v1.29 by the way?
You can find it here: http://weblookandfeel.com/downloads/prerelease/

**Comment by mgarin** [comment]
Tried under a few different OS/JDK versions - works fine everywhere.

Does this issue affect only table scrolling?
Or any other component also lags when scrolled with mouse?

If it is just table - it can be lagging because of:
- Heavy operations (for example client-server requests) made in table cell renderer
- Modified viewport scrolling mode (for example SIMPLE_SCROLL_MODE instead of default one)
- Insufficient RAM (for example if you run application under 30-20-10 mb or RAM or even less)

About available scroll modes in viewport (this is a Swing feature btw):
```java
    /**
      * Use <code>graphics.copyArea</code> to implement scrolling.
      * This is the fastest for most applications.
      *
      * @see #setScrollMode
      * @since 1.3
      */
    public static final int BLIT_SCROLL_MODE = 1;

    /**
      * Draws viewport contents into an offscreen image.
      * This was previously the default mode for <code>JTable</code>.
      * This mode may offer advantages over "blit mode"
      * in some cases, but it requires a large chunk of extra RAM.
      *
      * @see #setScrollMode
      * @since 1.3
      */
    public static final int BACKINGSTORE_SCROLL_MODE = 2;

    /**
      * This mode uses the very simple method of redrawing the entire
      * contents of the scrollpane each time it is scrolled.
      * This was the default behavior in Swing 1.0 and Swing 1.1.
      * Either of the other two options will provide better performance 
      * in most cases.
      *
      * @see #setScrollMode
      * @since 1.3
      */
    public static final int SIMPLE_SCROLL_MODE = 0;
```

If it is not table - issue might be somewhere else. For example there might be some global mouse wheel listener that does something heavy on each wheel movement. But seems that isn't the case if lags appear even on simple example you have provided above.

**Comment by mgarin** [comment]
Tried a few more configuration but still no success at reproducing this one.
Did you try using v1.29 pre-release binaries?

You can find them here: http://weblookandfeel.com/downloads/prerelease/
I have just updated them with a few bugfixes for other matters.

**Comment by oliviermorpa** [comment]
its working now since last prerealse.
thanks

**Comment by mgarin** [comment]
So I guess that was indeed something related with older version.

**Comment by oliviermorpa** [comment]
i open again this case because the scrolling is very slow and not working with mouse scroll when i put table in webtabbedPane.

**Comment by Sciss** [comment]
I agree that it's very slow. The same for `JScrollPane` wrapping for example a table.

**Comment by mgarin** [comment]
I finally realized that "slow scrolling" was referring to the amount of pixels scrolled per mouse wheel scroll operation and not some abstract interface lag while scrolling.

I have actually stumbled upon the root cause of the low scrolling speed in WebScrollPane and have added an appropriate fix in my local copy of `styling` branch. So the fix will be available together with all other changes as soon as `styling` branch is merged into `master`.
